### DISCUSSION TITLE
Option to decrease initial http-requests

### DISCUSSION THREAD

**Discussion by mbogdan0** [vercel/next.js#11994] [Help]
By default, the very simple NextJS application (after compilation) makes a dozen of http-requests: JS files, a lot of chunks, a bunch of css files and so on.

 NextJs is proud that it has zero configuration, but if you care about your app, first of all you must create `next.config.js` to do something to reduce http-requests.

I have googled a lot and all I found is something like
```
        config.plugins.push(
            new webpack.optimize.LimitChunkCountPlugin({
                maxChunks: 1
            })
        );
```
This supposed to reduce a number of chunks but it doesn't work as expected. I still have a dozen of chunks after that.

My question is:
1) How can I make my CSS _inline_? (to put all css in a _style_ tag)
2) How can I reduce http-requests and put all files in one JS file? (or at least several files)

Thank you

**Comment by hpachy** [comment]
Hello, 

I started since a week to learn next js and I have the same question. I'm currently stuck because of my SEO score which warn me to reduce my http-requests...

**Comment by kgabor545** [comment]
Same here. Any solution since then?

**Comment by johnxie** [comment]
+1 on this for https://www.taskade.com, as we are seeing 50+ requests on average for each page.

**Reply by sharifffffffffffff** [reply]
Same Here with - [medvarsity.com](https://medvarsity.com/)

**Comment by malaville** [comment]
+1 we are using amplify and cold lambdas are painfully serving these unecessarily chunked items; For example I always have webpack/framework/main chunks, on every page, there is no need to seperate these chunks.

**Reply by devajakamerus** [reply]
You can add to your next.config
`webpack: function (config, options) {
    config.optimization.splitChunks.minSize=50000; // size in bytes     
    return config;
}`

The bigger the size, the longer the build will take...

**Reply by mxswat** [reply]
@devajakamerus Are you sure this works? I tried it and it does not seem to do anything for me.
